File: odk/examples/DevelopersGuide/OfficeDev/Clipboard/python/clipboard_owner.py
# -*- tab-width: 4; indent-tabs-mode: nil; py-indent-offset: 4 -*-
#
# This file is part of the LibreOffice project.
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

import unohelper

from com.sun.star.datatransfer.clipboard import XClipboardOwner

# A simple clipboard owner for the clipboard.py example

class ClipboardOwner(unohelper.Base, XClipboardOwner):
    def __init__(self):
        self.is_owner = True

    def isClipboardOwner(self):
        return self.is_owner

    ''' XClipboardOwner methods '''
    def lostOwnership(self, x_clipboard, x_transferable):
        print("Lost clipboard ownership...")
        self.is_owner = False

# vim: set shiftwidth=4 softtabstop=4 expandtab:

File: odk/examples/DevelopersGuide/OfficeDev/Clipboard/ClipboardOwner.java
import com.sun.star.datatransfer.*;
import com.sun.star.datatransfer.clipboard.*;

//--------------------------
//  A simple clipboard owner
//--------------------------

public class ClipboardOwner implements XClipboardOwner
{
    public void lostOwnership( XClipboard xClipboard, XTransferable xTransferable )
    {
        System.out.println("");
        System.out.println( "Lost clipboard ownership..." );
        System.out.println("");

        isowner = false;
    }

    public boolean isClipboardOwner()
    {
        return isowner;
    }

    private boolean isowner = true;
}
